fst/en/inflection.py

```diff
@@ -0,0 +1,100 @@
+"""
+Inflects English nouns - creates a plural form from singular.
+
+Install Pynini package before running.
+"""
+
+import pynini as p
+
+from pynini.lib import pynutil
+from pynini.lib import rewrite
+
+def _priority_union(q: p.Fst, r: p.Fst, sigma: p.Fst) -> p.Fst:
+  complement_domain_q = sigma - p.project(q, 'input')
+  return p.union(q, complement_domain_q @ r)
+
+_v = p.union('a', 'e', 'i', 'o', 'u')
+_c = p.union('b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n',
+             'p', 'q', 'r', 's', 't', 'v', 'w', 'x', 'y', 'z')
+_sigma = p.union(_v, _c).closure().optimize()
+
+_exceptions = p.string_map([
+    # Zero plurals.
+    'deer',
+    'fish',
+    'sheep',
+    'series',
+    'species',
+    # Stem changes.
+    ('child', 'children'),
+    ('goose', 'geese'),
+    ('man', 'men'),
+    ('woman', 'women'),
+    ('tooth', 'teeth'),
+    ('foot', 'feet'),
+    ('mouse', 'mice'),
+    ('person', 'people'),
+    ('penny', 'pence'),
+    # Irregular suffixes.
+    ('child', 'children'),
+    ('ox', 'oxen'),
```

Comment:
This is a good start. The list can get much larger. You may want to make this list data driven instead of embedding it in code.

Reply:
All of the exceptions above and the rules would be dumped into an FST file that could be used from C++ (OpenFST). Exceptions wouldn't be in the code - we use python to build the rules and serialize the "model" to disk.
Of course, we can move this list to a file and load it instead of specifying in Python code. A refactor for later.

Reply:
Refactored exceptions to the file in data/ folder, but they'll still be part of the final FST once dumped.